test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule(
+        "Priority_1",
+        "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+        ""
+      );
+    } catch (error) {
+      errorWasThrown = true;
+      // should.equal(!error.message.search("Filter is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });*/
+  it("Removing non existing rule on a subscription that doesnt have any rules should throw error", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.removeRule("Priority_5");
+    } catch (error) {
+      should.equal(!error.message.search("could not be found"), false);
+      errorWasThrown = true;
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Removing non existing rule on a subscription that has other rules should throw error", async function(): Promise<
+    void
+  > {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 1");
+      await subscriptionClient.removeRule("Priority_5");
+    } catch (error) {
+      errorWasThrown = true;
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Subscription with 0/1/multiple rules returns rules as expected", async function(): Promise<
+    void
+  > {
+    let rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 0);
+
+    let expr = "(priority = 1)";
+    await subscriptionClient.addRule("Priority_1", expr);
+    rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+    should.equal(JSON.stringify(rules[0].filter), JSON.stringify({ expression: expr }));
+
+    expr = "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')";
+    await subscriptionClient.addRule(
+      "Priority_2",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')"
+    );
+    rules = await subscriptionClient.getRules();
+    should.equal(rules[1].name, "Priority_2");
+    should.equal(JSON.stringify(rules[0].filter), JSON.stringify({ expression: expr }));
+  });
+
+  it("Default rule is returned for the subscription for which no rules were added", async function(): Promise<
+    void
+  > {
+    const rules = await defaultSubscriptionClient.getRules();
+    should.equal(rules[0].name, "$Default");
+  });
+
+  it("Rule with SQL filter returns expected filter expression", async function(): Promise<void> {
+    const expr = "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')";
+    await subscriptionClient.addRule("Priority_2", expr);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_2");
+    should.equal(JSON.stringify(rules[0].filter), JSON.stringify({ expression: expr }));
+  });
+  /*
+  it("Rule with SQL filter and action returns expected filter and action expression", async function(): Promise<
```

Comment:
What happened here?

Reply:
As I've already mentioned, the action statement is not getting added in the rules for the subscription.